index.d.ts

```diff
@@ -0,0 +1,15 @@
+declare module "messageformat" {
+    class MessageFormat {
+        constructor(message: { [pluralFuncs: string]: Function });
+        constructor(message: string[]);
+        constructor(message: string);
+        addFormatters: (format: {}) => MessageFormat;
+        disablePluralKeyChecks: () => MessageFormat;
+        setBiDiSupport: (enable: boolean) => MessageFormat;
+        setStrictNumberSign: (enable: boolean) => MessageFormat;
+        compile: (messages: string, locale?: string) => Msg;
```

Comment:
This is missing the case of `messages` being an object, which is pretty important. Maybe something like this?
```ts
type SrcMessage = string | SrcObject;

interface SrcObject {
    [x: string]: SrcMessage;
}

compile: (messages: SrcMessage, locale?: string) => Msg;
```